core/app/[locale]/(default)/(faceted)/search/page.tsx

```diff
@@ -5,332 +5,261 @@ import { cache } from 'react';
-    { value: 'best_reviewed', label: t('byReview') },
-    { value: 'lowest_price', label: t('priceAscending') },
-    { value: 'highest_price', label: t('priceDescending') },
-    { value: 'relevance', label: t('relevance') },
-  ];
+interface Props {
+  params: Promise<{ locale: string }>;
+  searchParams: Promise<Record<string, string | string[] | undefined>>;
 }
 
-async function getPaginationInfo(props: Props): Promise<CursorPaginationInfo> {
-  const searchTerm = await getSearchTerm(props);
-
-  if (searchTerm === '') {
-    return {
-      startCursorParamName: 'before',
-      endCursorParamName: 'after',
-      endCursor: null,
-      startCursor: null,
-    };
-  }
+export async function generateMetadata({ params }: Props): Promise<Metadata> {
+  const { locale } = await params;
 
-  const search = await getSearch(props);
-  const { hasNextPage, hasPreviousPage, endCursor, startCursor } = search.products.pageInfo;
+  const t = await getTranslations({ locale, namespace: 'Faceted.Search' });
 
   return {
-    startCursorParamName: 'before',
-    endCursorParamName: 'after',
-    endCursor: hasNextPage ? endCursor : null,
-    startCursor: hasPreviousPage ? startCursor : null,
+    title: t('title'),
   };
 }
 
-async function getShowCompare() {
-  const data = await getSearchPageData();
+export default async function Search(props: Props) {
+  const { locale } = await props.params;
 
-  return data.settings?.storefront.catalog?.productComparisonsEnabled ?? false;
-}
+  setRequestLocale(locale);
 
-const cachedCompareProductIds = cache(async (props: Props) => {
-  const searchParams = await props.searchParams;
+  const t = await getTranslations('Faceted');
+
+  const { settings } = await getSearchPageData();
+
+  const productComparisonsEnabled =
+    settings?.storefront.catalog?.productComparisonsEnabled ?? false;
+
+  const createSearchSearchParamsCache = cache(async () => {
+    const searchParams = await props.searchParams;
+    const searchTerm = typeof searchParams.term === 'string' ? searchParams.term : '';
+
+    if (!searchTerm) {
+      return null;
+    }
+
+    const search = await fetchFacetedSearch(searchParams);
+    const searchFacets = search.facets.items;
+    const transformedSearchFacets = await facetsTransformer({
+      refinedFacets: searchFacets,
+      allFacets: searchFacets,
+      searchParams: {},
+    });
+    const searchFilters = transformedSearchFacets.filter((facet) => facet != null);
+    const filterParsers = getFilterParsers(searchFilters);
+
+    // If there are no filters, return `null`, since calling `createSearchParamsCache` with an empty
+    // object will throw the following cryptic error:
+    //
+    // ```
+    // Error: [nuqs] Empty search params cache. Search params can't be accessed in Layouts.
+    //   See https://err.47ng.com/NUQS-500
+    // ```
+    if (Object.keys(filterParsers).length === 0) {
+      return null;
+    }
+
+    return createSearchParamsCache(filterParsers);
+  });
```

Comment:
We should replace this with a loader and instead of a function, we could create a loader and parse the search params within a `Streamable.from` so that we have a `streamableParsedSearchParams` we use instead of `props.searchParam` throughout the component